### DISCUSSION TITLE
Feature Request: Deploy library pdbs to build output

### DISCUSSION THREAD

**Discussion by IRebo** [microsoft/vcpkg#17808] [Ideas]
Would it be possible to extend the vcpkg nuget in a way that it deploys a native dll along with its pdb to build output ? Currently it seems to deploy the necessary .dlls but the pdbs are not copied. This makes things more complicated if we want to package the complete build output.

I saw looking at https://github.com/microsoft/vcpkg/blob/master/scripts/buildsystems/msbuild/applocal.ps1.
My assumption is that extending the deployBinary function would do the trick ? Is there a reason why its not done already ?

thanks
Ivan

**Comment by sylveon** [comment]
I would also like to get PDBs copied to the output, so that it can be automatically copied by VS into my .appx package

**Comment by BillyONeal** [comment]
> I would also like to get PDBs copied to the output, so that it can be automatically copied by VS into my .appx package

Generally speaking PDBs should not be deployed on end-user machines so they shouldn't be going into an appx.

We may still want to look at doing something like this someday but it's mostly applicable to folks who are shipping static libraries rather than applications.

To truly deliver an end to end experience here we might need to package the sources too.

**Comment by sylveon** [comment]
The way `.appx` works is weird.
When building for local or remote testing, everything gets into a single `.appx`, PDB included.
But when packaging and uploading, VS uses an `.appxbundle` file, and this archive contains both an `.appx` and `.appxsym` file.
VS does some ✨ magic ✨ from the normal build output for that purpose, so vcpkg simply copying PDBs along with the DLLs should work.

`.appxsym` has symbols, and those will be used in telemetry data.
The `.appx` file is what consumers will end up downloading, and doesn't have symbols.

Without symbols, stack traces obtained from the Microsoft Store telemetry will lack any kind of function info:
![image](https://user-images.githubusercontent.com/6440374/101226130-7fdf4b80-3661-11eb-993a-62ecec283372.png)

**Comment by BillyONeal** [comment]
I see, I didn't realize there was some postprocessing step done by the store to strip those out.

Can we assume that for a given DLL `x.dll` the PDB will be named `x.pdb`?

**Comment by sylveon** [comment]
Yes

**Comment by nyoma-diamond** [comment]
Apologies for the necropost, but this would be very useful for debug builds! 

Currently when using vcpkg in manifest mode it sometimes isn't possible to debug issues coming from inside dependencies because the PDBs don't get deployed alongside the DLLs in the build (particularly in cases where the dependency is installed from the binary cache instead of `$VCPKG_ROOT`). This would be purely for development and never make its way to end-users, but would still be the same deployment behavior.

See: https://github.com/microsoft/vcpkg/issues/32192#issuecomment-4355592351
